### PR TITLE
refactor(p2p): rename peer_id.PeerId to peer.Peer

### PR DESCRIPTION
**Pull request by jansegre**
### Motivation

This is the first part of a series of refactors. This first refactor is a step to make using `PeerId` for only the `id` part of `Peer`.

### Acceptance Criteria

- rename `hathor.p2p.peer_id.PeerId` to `hathor.p2p.peer.Peer`

### Checklist

- [x] If you are requesting a merge into `master`, confirm this code is production-ready and can be included in future releases as soon as it gets merged

### CODE REVIEW

File: hathor/manager.py [hunk 1]
```diff
@@ -96,7 +96,7 @@ def __init__(
         pubsub: PubSubManager,
         consensus_algorithm: ConsensusAlgorithm,
         daa: DifficultyAdjustmentAlgorithm,
-        peer_id: PeerId,
+        peer_id: Peer,
```
Comment (msbrogli) [inline hathor/manager.py:99]: Same here.

File: tests/p2p/test_sync_v2.py [hunk 2]
```diff
@@ -311,7 +311,7 @@ def custom_gen_new_tx(manager: HathorManager, _address: str, value: int) -> Tran
         self.assertGreater(mempool_tips_count, 30)
 
         # Create a new peer and run sync for a while (but stop before getting synced).
-        peer_id = PeerId()
+        peer_id = Peer()
```
Comment (msbrogli) [inline tests/p2p/test_sync_v2.py:314]: Same here.

File: tests/simulation/test_simulator_itself.py [hunk 3]
```diff
@@ -42,7 +42,7 @@ def tearDown(self) -> None:
     def create_simulator_peer(
         self,
         simulator: Simulator,
-        peer_id_pool: list[PeerId],
+        peer_id_pool: list[Peer],
```
Comment (msbrogli) [inline tests/simulation/test_simulator_itself.py:45]:
```suggestion
        peer_pool: list[Peer],
```

File: tests/unittest.py [hunk 4]
```diff
@@ -40,7 +40,7 @@ def short_hashes(container: Collection[bytes]) -> Iterable[str]:
     return map(lambda hash_bytes: hash_bytes[-2:].hex(), container)
 
 
-def _load_peer_id_pool(file_path: Optional[str] = None) -> Iterator[PeerId]:
+def _load_peer_id_pool(file_path: Optional[str] = None) -> Iterator[Peer]:
```
Comment (msbrogli) [inline tests/unittest.py:43]:
```suggestion
def _load_peer_pool(file_path: Optional[str] = None) -> Iterator[Peer]:
```

File: tests/unittest.py [hunk 5]
```diff
@@ -135,11 +135,11 @@ def tearDown(self) -> None:
     def reset_peer_id_pool(self) -> None:
         self._free_peer_id_pool = self.new_peer_id_pool()
 
-    def new_peer_id_pool(self) -> list[PeerId]:
+    def new_peer_id_pool(self) -> list[Peer]:
```
Comment (msbrogli) [inline tests/unittest.py:138]:
```suggestion
    def new_peer_pool(self) -> list[Peer]:
```

File: tests/unittest.py [hunk 6]
```diff
@@ -135,11 +135,11 @@ def tearDown(self) -> None:
...
         return PEER_ID_POOL.copy()
 
-    def get_random_peer_id_from_pool(self, pool: Optional[list[PeerId]] = None,
-                                     rng: Optional[Random] = None) -> PeerId:
+    def get_random_peer_id_from_pool(self, pool: Optional[list[Peer]] = None,
```
Comment (msbrogli) [inline tests/unittest.py:141]:
```suggestion
    def get_random_peer_from_pool(self, pool: Optional[list[Peer]] = None,
```

File: tests/unittest.py [hunk 7]
```diff
@@ -194,7 +194,7 @@ def create_peer_from_builder(self, builder: Builder, start_manager: bool = True)
     def create_peer(  # type: ignore[no-untyped-def]
         self,
         network: str,
-        peer_id: PeerId | None = None,
+        peer_id: Peer | None = None,
```
Comment (msbrogli) [inline tests/unittest.py:197]:
```suggestion
        peer: Peer | None = None,
```

File: tests/unittest.py [hunk 8]
```diff
@@ -226,7 +226,7 @@ def create_peer(  # type: ignore[no-untyped-def]
             builder.set_pubsub(pubsub)
 
         if peer_id is None:
-            peer_id = PeerId()
+            peer_id = Peer()
         builder.set_peer_id(peer_id)
```
Comment (msbrogli) [inline tests/unittest.py:230]:
```suggestion
            peer = Peer()
        builder.set_peer(peer)
```

File: tests/p2p/netfilter/test_match.py [hunk 9]
```diff
@@ -202,8 +202,8 @@ class BaseNetfilterMatchTest(unittest.TestCase):
 
     def test_match_peer_id(self) -> None:
         network = 'testnet'
-        peer_id1 = PeerId()
-        peer_id2 = PeerId()
+        peer_id1 = Peer()
+        peer_id2 = Peer()
```
Comment (msbrogli) [inline tests/p2p/netfilter/test_match.py:206]:
```suggestion
        peer_1 = Peer()
        peer_2 = Peer()
```

File: hathor/builder/builder.py [hunk 10]
```diff
@@ -96,7 +96,7 @@ class StorageType(Enum):
 
 class BuildArtifacts(NamedTuple):
     """Artifacts created by a builder."""
-    peer_id: PeerId
+    peer_id: Peer
```
[the remark this answers is not in the document]
Comment (jansegre) [reply hathor/builder/builder.py]: Done.

File: hathor/builder/builder.py [hunk 11]
```diff
@@ -137,7 +137,7 @@ def __init__(self) -> None:
         self._checkpoints: Optional[list[Checkpoint]] = None
         self._capabilities: Optional[list[str]] = None
 
-        self._peer_id: Optional[PeerId] = None
+        self._peer_id: Optional[Peer] = None
```
[the remark this answers is not in the document]
Comment (jansegre) [reply hathor/builder/builder.py]: Done.

File: hathor/builder/builder.py [hunk 12]
```diff
@@ -361,7 +361,7 @@ def _get_soft_voided_tx_ids(self) -> set[bytes]:
 
         return set(settings.SOFT_VOIDED_TX_IDS)
 
-    def _get_peer_id(self) -> PeerId:
+    def _get_peer_id(self) -> Peer:
```
[the remark this answers is not in the document]
Comment (jansegre) [reply hathor/builder/builder.py]: Done.

File: hathor/builder/cli_builder.py [hunk 13]
```diff
@@ -98,7 +98,7 @@ def create_manager(self, reactor: Reactor) -> HathorManager:
         self.log = logger.new()
         self.reactor = reactor
 
-        peer_id = PeerId.create_from_json_path(self._args.peer) if self._args.peer else PeerId()
+        peer_id = Peer.create_from_json_path(self._args.peer) if self._args.peer else Peer()
```
[the remark this answers is not in the document]
Comment (jansegre) [reply hathor/builder/cli_builder.py]: Done.

[the remark this answers is not in the document]
Comment (jansegre) [reply hathor/builder/cli_builder.py]: I'll do that one on a separate PR. Since it's not exactly a refactor.

File: hathor/cli/peer_id.py [hunk 14]
```diff
@@ -12,17 +12,17 @@
...
 
 import json
 
 
 def main() -> None:
-    from hathor.p2p.peer_id import PeerId
+    from hathor.p2p.peer import Peer
 
-    peer_id = PeerId()
+    peer_id = Peer()
```
[the remark this answers is not in the document]
Comment (jansegre) [reply hathor/cli/peer_id.py]: Done.

File: hathor/manager.py (hunk 1 above)
Comment (jansegre) [reply hathor/manager.py]: Done.

File: tests/event/event_simulation_tester.py [hunk 15]
```diff
@@ -34,7 +34,7 @@ class BaseEventSimulationTester(SimulatorTestCase):
     builder: Builder
 
     def _create_artifacts(self) -> None:
-        peer_id = PeerId()
+        peer_id = Peer()
```
[the remark this answers is not in the document]
Comment (jansegre) [reply tests/event/event_simulation_tester.py]: Done.

File: tests/p2p/test_protocol.py [hunk 16]
```diff
@@ -19,8 +19,8 @@ class BaseHathorProtocolTestCase(unittest.TestCase):
     def setUp(self) -> None:
         super().setUp()
         self.network = 'testnet'
-        self.peer_id1 = PeerId()
-        self.peer_id2 = PeerId()
+        self.peer_id1 = Peer()
```
[the remark this answers is not in the document]
Comment (jansegre) [reply tests/p2p/test_protocol.py]: Done.

File: tests/p2p/test_sync_v2.py [hunk 17]
```diff
@@ -66,7 +66,7 @@ def _run_restart_test(self, *, full_verification: bool, use_tx_storage_cache: bo
 
         # Create a new peer and run sync for a while (but stop before getting synced).
         path = self.mkdtemp()
-        peer_id = PeerId()
+        peer_id = Peer()
```
[the remark this answers is not in the document]
Comment (jansegre) [reply tests/p2p/test_sync_v2.py]: Done.

File: tests/p2p/test_sync_v2.py [hunk 18]
```diff
@@ -220,7 +220,7 @@ def test_exceeds_streaming_and_mempool_limits(self) -> None:
             print()
 
         # Create a new peer and run sync for a while (but stop before getting synced).
-        peer_id = PeerId()
+        peer_id = Peer()
```
[the remark this answers is not in the document]
Comment (jansegre) [reply tests/p2p/test_sync_v2.py]: Done.

File: tests/p2p/test_sync_v2.py (hunk 2 above)
Comment (jansegre) [reply tests/p2p/test_sync_v2.py]: Done.

File: tests/simulation/test_simulator_itself.py (hunk 3 above)
Comment (jansegre) [reply tests/simulation/test_simulator_itself.py]: Done.

File: tests/unittest.py (hunk 4 above)
Comment (jansegre) [reply tests/unittest.py]: Done.

File: tests/unittest.py (hunk 5 above)
Comment (jansegre) [reply tests/unittest.py]: Done.

File: tests/unittest.py (hunk 6 above)
Comment (jansegre) [reply tests/unittest.py]: Done.

File: tests/unittest.py (hunk 8 above)
Comment (jansegre) [reply tests/unittest.py]: Done.

File: tests/p2p/netfilter/test_match.py (hunk 9 above)
Comment (jansegre) [reply tests/p2p/netfilter/test_match.py]: Done.

File: tests/cli/test_sysctl_init.py [hunk 19]
```diff
@@ -195,6 +197,6 @@ def register_signal_handlers(self) -> None:
         self.assertTrue(run_node is not None)
         conn = run_node.manager.connections
 
-        curr_always_enabled_sync = list(conn.always_enable_sync)
+        curr_always_enabled_sync = list(map(str, conn.always_enable_sync))
```
Comment (glevco) [inline tests/cli/test_sysctl_init.py:200]: Why this change?

**Review by glevco — APPROVED** [review]

File: tests/cli/test_sysctl_init.py (hunk 19 above)
Comment (jansegre) [reply tests/cli/test_sysctl_init.py:200]: I had to make this for the test to work, because the assertion bellow needs sets of the same type, and `conn.always_enable_sync` is a `set[PeerId]`, but now that you mention it I think this is only changed in the next PR, so I might have backported this fix too early.

Comment (glevco) [reply tests/cli/test_sysctl_init.py:200]: Ok!

File: hathor/builder/builder.py (hunk 10 above)
[the remark this answers is not in the document]
Comment (msbrogli) [reply hathor/builder/builder.py]: Thanks!

File: hathor/builder/builder.py (hunk 11 above)
[the remark this answers is not in the document]
Comment (msbrogli) [reply hathor/builder/builder.py]: Thanks!

File: hathor/builder/builder.py [hunk 20]
```diff
@@ -337,7 +337,7 @@ def set_capabilities(self, capabilities: list[str]) -> 'Builder':
         self._capabilities = capabilities
         return self
 
-    def set_peer_id(self, peer_id: PeerId) -> 'Builder':
+    def set_peer_id(self, peer_id: Peer) -> 'Builder':
```
[the remark this answers is not in the document]
Comment (msbrogli) [reply hathor/builder/builder.py]: Thanks!

File: hathor/builder/builder.py (hunk 12 above)
[the remark this answers is not in the document]
Comment (msbrogli) [reply hathor/builder/builder.py]: Thanks!

File: hathor/builder/cli_builder.py (hunk 13 above)
[the remark this answers is not in the document]
Comment (msbrogli) [reply hathor/builder/cli_builder.py]: Thanks!

File: hathor/cli/peer_id.py (hunk 14 above)
[the remark this answers is not in the document]
Comment (msbrogli) [reply hathor/cli/peer_id.py]: Thanks!

File: hathor/manager.py (hunk 1 above)
Comment (msbrogli) [reply hathor/manager.py]: Thanks!

File: tests/event/event_simulation_tester.py (hunk 15 above)
[the remark this answers is not in the document]
Comment (msbrogli) [reply tests/event/event_simulation_tester.py]: Thanks!

File: tests/p2p/test_protocol.py (hunk 16 above)
[the remark this answers is not in the document]
Comment (msbrogli) [reply tests/p2p/test_protocol.py]: Thanks!

File: tests/p2p/test_sync_v2.py (hunk 17 above)
[the remark this answers is not in the document]
Comment (msbrogli) [reply tests/p2p/test_sync_v2.py]: Thanks!

File: tests/p2p/test_sync_v2.py (hunk 18 above)
[the remark this answers is not in the document]
Comment (msbrogli) [reply tests/p2p/test_sync_v2.py]: Thanks!

File: tests/p2p/test_sync_v2.py (hunk 2 above)
Comment (msbrogli) [reply tests/p2p/test_sync_v2.py]: Thanks!

File: tests/simulation/test_simulator_itself.py (hunk 3 above)
Comment (msbrogli) [reply tests/simulation/test_simulator_itself.py]: Thanks!

File: tests/unittest.py (hunk 4 above)
Comment (msbrogli) [reply tests/unittest.py]: Thanks!

File: tests/unittest.py (hunk 5 above)
Comment (msbrogli) [reply tests/unittest.py]: Thanks!

File: tests/unittest.py (hunk 6 above)
Comment (msbrogli) [reply tests/unittest.py]: Thanks!

File: tests/unittest.py (hunk 7 above)
Comment (msbrogli) [reply tests/unittest.py]: Thanks!

File: tests/unittest.py (hunk 8 above)
Comment (msbrogli) [reply tests/unittest.py]: Thanks!

File: tests/p2p/netfilter/test_match.py (hunk 9 above)
Comment (msbrogli) [reply tests/p2p/netfilter/test_match.py]: Thanks!

File: hathor/builder/cli_builder.py (hunk 13 above)
[the remark this answers is not in the document]
Comment (msbrogli) [reply hathor/builder/cli_builder.py]: Ok. Thanks!

**Review by msbrogli — APPROVED** [review]
